Mathlib/RingTheory/KrullDimension/Zero.lean

```diff
@@ -0,0 +1,215 @@
+  tfae_have 3 → 4 := by
+    intro H
+    refine ⟨?_, ?_⟩
+    · intro e
+      obtain ⟨n, hn⟩ := (Ideal.eq_top_iff_one _).mp e
+      exact (H 0).mp .zero ((show (1 : R) = 0 by simpa using hn) ▸ isUnit_one)
+    · intro I hI
+      obtain ⟨x, hx, hx'⟩ := (SetLike.lt_iff_le_and_exists.mp hI).2
+      exact Ideal.eq_top_of_isUnit_mem _ hx (not_not.mp ((H x).not.mp hx'))
+  tfae_have 4 → 2 := by
+    intro H
+    exact ⟨_, H.isPrime, fun p (hp : p.IsPrime) ↦
+      (H.eq_of_le hp.ne_top (nilradical_le_prime p)).symm⟩
+  tfae_have 2 → 1 := by
+    rintro ⟨P, hP₁, hP₂⟩
+    obtain ⟨P, hP₃, -⟩ := P.exists_le_maximal hP₁.ne_top
+    obtain rfl := hP₂ P hP₃.isPrime
+    exact ⟨.mk₀ fun Q h ↦ hP₂ Q h ▸ hP₃, .of_unique_max_ideal ⟨P, hP₃, fun Q h ↦ hP₂ Q h.isPrime⟩⟩
+  tfae_finish
+
+@[simp]
+lemma le_isUnit_iff_zero_not_mem [IsLocalRing R]
+    {M : Submonoid R} : M ≤ IsUnit.submonoid R ↔ 0 ∉ M := by
+  have := ((Ring.krullDimLE_zero_and_isLocalRing_tfae R).out 0 2 rfl rfl).mp ⟨‹_›, ‹_›⟩
+  exact ⟨fun h₁ h₂ ↦ not_isUnit_zero (h₁ h₂),
+    fun H x hx ↦ (this x).not_left.mp fun ⟨n, hn⟩ ↦ H (hn ▸ pow_mem hx n)⟩
+
+variable (R) in
+theorem Ring.KrullDimLE.existsUnique_isPrime [IsLocalRing R] :
+    ∃! I : Ideal R, I.IsPrime :=
+  ((Ring.krullDimLE_zero_and_isLocalRing_tfae R).out 0 1 rfl rfl).mp ⟨‹_›, ‹_›⟩
+
+theorem Ring.KrullDimLE.eq_maximalIdeal_of_isPrime [IsLocalRing R] (J : Ideal R) [J.IsPrime] :
+    J = IsLocalRing.maximalIdeal R :=
+  (((Ring.krullDimLE_zero_and_isLocalRing_tfae R).out 0 1 rfl rfl).mp ⟨‹_›, ‹_›⟩).unique
+    ‹_› inferInstance
+
+theorem Ring.KrullDimLE.isNilpotent_iff_mem_maximalIdeal [IsLocalRing R] {x} :
+    IsNilpotent x ↔ x ∈ IsLocalRing.maximalIdeal R :=
+  ((Ring.krullDimLE_zero_and_isLocalRing_tfae R).out 0 2 rfl rfl).mp ⟨‹_›, ‹_›⟩ x
+
+theorem Ring.KrullDimLE.isNilpotent_iff_mem_nonunits [IsLocalRing R] {x} :
+    IsNilpotent x ↔ x ∈ nonunits R :=
+  isNilpotent_iff_mem_maximalIdeal
+
+variable (R) in
+theorem Ring.KrullDimLE.nilradical_eq_maximalIdeal [IsLocalRing R] :
+    nilradical R = IsLocalRing.maximalIdeal R :=
+  Ideal.ext fun _ ↦ isNilpotent_iff_mem_maximalIdeal
+
+omit [Ring.KrullDimLE 0 R] in
+variable (R) in
+theorem IsLocalRing.of_isMaximal_nilradical [(nilradical R).IsMaximal] :
+    IsLocalRing R :=
+  (((Ring.krullDimLE_zero_and_isLocalRing_tfae R).out 3 0 rfl rfl).mp ‹_›).2
+
+omit [Ring.KrullDimLE 0 R] in
+variable (R) in
+theorem Ring.KrullDimLE.of_isMaximal_nilradical [(nilradical R).IsMaximal] :
+    Ring.KrullDimLE 0 R :=
+  (((Ring.krullDimLE_zero_and_isLocalRing_tfae R).out 3 0 rfl rfl).mp ‹_›).1
+
+omit [Ring.KrullDimLE 0 R] in
+lemma Ring.KrullDimLE.of_isLocalization (p : Ideal R) (hp : p ∈ minimalPrimes R)
+    (S : Type*) [CommSemiring S] [Algebra R S] [IsLocalization.AtPrime S p (hp := hp.1.1)] :
+    Ring.KrullDimLE 0 S :=
+  have := IsLocalization.subsingleton_primeSpectrum_of_mem_minimalPrimes p hp S
+  ⟨Order.krullDim_nonpos_of_subsingleton⟩
```

Comment:
```suggestion
  inferInstance
```
(untested)

Reply:
Sadly it doesn't. Maybe it needs the instance in your other PR.

Reply:
I see.